libtenzir/include/tenzir/detail/adaptive_table_slice_builder_guards.hpp

```diff
@@ -25,6 +25,9 @@ struct type_from_data
 template <class T>
 using type_from_data_t = typename type_from_data<T>::type;
 
+using parent_record_builder_provider
+  = std::function<concrete_series_builder<record_type>*()>;
```

Comment:
Please document the contract of this type.

Reply:
done in ce71c24dd38c9e8d377d0c6c42b1c6c5bbc1f165